### PR TITLE
Added descriptions

### PR DESCRIPTION
**Pull request by AllThingsSmitty**
Related to issue #928

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -86,10 +86,10 @@
 ## Programming Languages
 
 - [JavaScript](https://github.com/sorrycc/awesome-javascript)
-	- [Promises](https://github.com/wbinnssmith/awesome-promises)
+	- [Promises](https://github.com/wbinnssmith/awesome-promises) - Resources for JavaScript Promises.
```
Comment (sindresorhus) [inline readme.md]: This says nothing new. Think about it. Awesome lists are lists of resources. The name is `Promises` and it's a sub-item of `JavaScript`.

Also note:

> Include a short description about the project/theme of the list. It should not describe the list itself. - https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md

Same applies to all the other descriptions here.

Comment (AllThingsSmitty) [inline readme.md]: Okay, let me go back and see what I can do.

Comment (legodude17) [inline readme.md]: I would suggest something like `New method for async in JavaScript`.

Comment (AllThingsSmitty) [inline readme.md]: @legodude17 I don't think that meets the requirements outlined in https://github.com/sindresorhus/awesome/issues/928. At some point it stops being "new".

Comment (legodude17) [inline readme.md]: @AllThingsSmitty How about "Different" then?

Comment (AllThingsSmitty) [inline readme.md]: @legodude17 maybe. I think it's going to be a challenge.

**Pull request closed without merging**